packages/salesforcedx-vscode-core/src/commands/dataQuery.ts

```diff
@@ -96,3 +218,210 @@ export const dataQuery = (): void => {
   const commandlet = new SfCommandlet(workspaceChecker, parameterGatherer, new DataQueryExecutor());
   void commandlet.run();
 };
+
+/**
+ * Generates table output from query records
+ */
+export const generateTableOutput = (records: any[], title: string): string => {
+  if (!records || records.length === 0) {
+    return '';
+  }
+
+  // Ensure the first record exists and is an object
+  const firstRecord = records[0];
+  if (!firstRecord || typeof firstRecord !== 'object') {
+    return '';
+  }
+
+  const fields = Object.keys(firstRecord).filter(key => key !== 'attributes');
+  // If no fields after filtering attributes, return empty string
+  if (fields.length === 0) {
+    return '';
+  }
+
+  const columns: Column[] = fields.map(field => ({
+    key: field,
+    label: field
+  }));
+  const rows: Row[] = records.map((record: { [x: string]: any }) => {
+    if (!record || typeof record !== 'object') {
+      return {};
+    }
+    return Object.fromEntries(fields.map(field => [field, formatFieldValueForDisplay(record[field])]));
+  });
+
+  return new Table().createTable(rows, columns, title);
+};
+
+/**
+ * Converts query records to CSV format
+ */
+export const convertToCSV = (records: any[]): string => {
+  if (!records || records.length === 0) {
+    return '';
+  }
+
+  // Ensure the first record exists and is an object
+  const firstRecord = records[0];
+  if (!firstRecord || typeof firstRecord !== 'object') {
+    return '';
+  }
+
+  const fields = Object.keys(firstRecord).filter(key => key !== 'attributes');
+  // If no fields after filtering attributes, return empty string
+  if (fields.length === 0) {
+    return '';
+  }
+
+  const header = fields.map(field => escapeCSVField(field)).join(',');
+  const rows = records.map((record: { [x: string]: any }) => {
+    if (!record || typeof record !== 'object') {
+      return '';
+    }
+    return fields
+      .map(field => {
+        const value = record[field];
+        return escapeCSVField(formatFieldValue(value));
+      })
+      .join(',');
+  });
+
```

Comment:
as is, it would leave a blank row.  I couldn't reproduce this by getting bad data from the server and I don't think it's a real scenario, so I just skipped that row instead of leaving it blank